### DISCUSSION TITLE
Use with custom Tileserver

### DISCUSSION THREAD

**Discussion by littlejustinh** [rnmapbox/maps#1497] [Q&A]
Is anyone out there using this library against a custom Tileserver? We use tileserver to host some data that I can't put on Mapbox, and I'm struggling big time for some reason to get layers to show up on a react native app, though we don't have any issues on the web. I am certain that it should work just fine, but I'm just hoping that someone can confirm that it should be working. 

I only have a very simple app for testing purposes so far, here is a snippet, though i did change the url and ids for privacy:

<MapboxGL.VectorSource
            id="vector_source_id"
            url="https://companydomain/tileserver_layer_id.json"
            attribution="my-app">
       <MapboxGL.LineLayer
              id="vector_layer_id"
              sourceId="vector_source_id"
              sourceLayerId="tileserver_layer_id"
         />
</MapboxGL.VectorSource>

This is the tileserver we use: https://github.com/maptiler/tileserver-gl

This is the trimmed down and string replaced version of the tilejson:

`{
  "tiles": [
    "https://companydomain/tileserver_layer_id/{z}/{x}/{y}.pbf"
  ],
  "name": "tileserver_layer_id.mbtiles",
  "format": "pbf",
  "basename": "tileserver_layer_id.mbtiles",
  "id": "tileserver_layer_id",
  "description": "tileserver_layer_id.mbtiles",
  "version": "2",
  "minzoom": 0,
  "maxzoom": 14,
  "center": [
    -99.503174,
    27.518015,
    14
  ],
  "bounds": [
    -104.52406,
    25.841551,
    -78.440794,
    42.429475
  ],
  "type": "overlay",
  "generator": "tippecanoe v1.36.0",
  "vector_layers": [{
    "id": "tileserver_layer_id",
    "description": "",
    "minzoom": 0,
    "maxzoom": 14,
    "fields": {
      ...
    }
  }],
  "tilestats": {},
  "tilejson": "2.0.0"
}
`

**Comment by ferdicus** [comment]
Do you have a turnkey solution to set up a local tileserver? Or host one somewhere, that'd make it easier to test this. 
I've not been using a local tile server, however I've been using HERE maps vector maps as described [here](https://developer.here.com/documentation/examples/vector-tiles/rendering/mapbox)

What I've done:
* download the style.json file via downloading the JSON provided when following this link: https://assets.vector.hereapi.com/styles/berlin/base/mapbox/tilezen?apikey=APIKEY (notice, that you can make a few request without being restricted, even with a non valid key, if you want to see the structure of the json 🤷🏿‍♂️ ).
* use the path to the `style.json` in `MapView`s `styleUrl`
